sdk/src/fee.rs

```diff
@@ -80,17 +80,9 @@ impl FeeStructure {
     pub fn calculate_fee(
         &self,
         message: &SanitizedMessage,
-        lamports_per_signature: u64,
         budget_limits: &FeeBudgetLimits,
         include_loaded_account_data_size_in_fee: bool,
     ) -> u64 {
-        // Fee based on compute units and signatures
-        let congestion_multiplier = if lamports_per_signature == 0 {
-            0.0 // test only
-        } else {
-            1.0 // multiplier that has no effect
-        };
-
         let signature_fee = message
             .num_signatures()
             .saturating_mul(self.lamports_per_signature);
```

Comment:
while removing "no-longer necessary" `congestion_multipler`, realized there might be a bug in this line 88 above. The signature_fee is always calculated with current bank's fee_structure's lamports_per_signature, it should be using passed in parameter `lamports_per_signature`, which is determined differently for nonce tx and normal tx. 

Isn't this a bug? @t-nelson @jstarry 

I traced it back to https://github.com/solana-labs/solana/pull/23095 where behavior seems changed.

Reply:
mmm, i don't remember the intent of this "congestion" fee. `lamports_per_signature` already contains the signature rate based congestion fee. it's calculated at the start of the slot, https://github.com/jackcmay/solana/blob/176ef5aeb25d7b3f331a12655654e801fed06235/runtime/src/bank.rs#L1563-L1578.

Reply:
@t-nelson can you comment on why durable nonces store `lamports_per_signature` and if we still intend to use that value in the future. The main thing we need to figure out to accept or reject this PR is whether or not we should be using the nonce-stored `lamports_per_signature` value from a previous bank here instead of the one calculated by the current bank.

Reply:
durable nonces store `lamports_per_signature` because some opinionated people were very adamant about preserving the "fee-payer pays deterministic fees" meme. afaik we've already unhooked them and just use fee rate from time of execution

Reply:
another way to look at it: if #34754 is legit bugfix?